admin/listPurchases/listView.html

```diff
@@ -185,6 +173,18 @@
     updateSubpageInputBox();
 }
 
+function onClickSubpageSelectorGeneric(e, modifyCondition, modifyCurrentPageIndex){
+    e.preventDefault();
+    if(!modifyCurrentPageIndex){
+        modifyCurrentPageIndex = modifyCondition;
+        modifyCondition = true;
+    }
+    if(modifyCondition){
+        modifyCurrentPageIndex()
+    }
```

Comment:
Having `modifyCondition` in here muddies the code and saves you on complexity where you call this function. Remove it, and use simple `if` statements where you call this function instead.

Don't reinvent the wheel where good syntax already exists.

E.g. the above:

```js
onClickSubpageSelectorGeneric(e, currentPageIndex + 1 < amountOfSubpages, () => currentPageIndex++)
```

should just be:

```js
if(currentPageIndex + 1 < amountOfSubpages) {
  onClickSubpageSelectorGeneric(e, () => currentPageIndex++)
}
```

That communicates *much more clearly* what is happening!

Reply:
*And* this lets you remove all the code for handling when no modifyCondition is passed to this function, making the function simpler too!